src/hooks/useAdvanceStep.js

```diff
@@ -0,0 +1,34 @@
+import { useRef, useState } from 'react'
+
+function useAdvanceStep() {
+  const ref = useRef(null)
+
+  const [currentStep, setCurrentStep] = useState(0)
+
+  const scrollTop = () => window.scrollTo({ x: 0 })
+
+  const changeTitle = title => (document.title = title)
```

Comment:
No local dependecies, so they can be moved outside of the component (and not get redefined on every update)